hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java

```diff
@@ -721,7 +723,13 @@ public String pathToKey(Path path) {
       path = new Path(workingDir, path);
     }
     // removing leading '/' char
-    String key = path.toUri().getPath().substring(1);
+    String key = path.toUri().getPath();
+
+    if (OzoneFSUtils.isValidName(key)) {
+      key = path.toUri().getPath().substring(1);
```

Comment:
Can we just do this here, instead of toUri().getPath() again?
`    LOG.trace("path for key:{} is:{}", key, path);
    return key.substring(1);
`

Reply:
Done.